### DISCUSSION TITLE
Error Chrome Driver di Selenium

### DISCUSSION THREAD

**Discussion by myibra** [teamalgoritma/community#76] [Q&A] [answered]
Selamat Siang Semuanya,

saya sedang belajar untuk web scraping menggunakan package selenium, tetapi pada ingin memanggil chromedriver nya error. berikut saya lampirkan untuk code dan error message nya.

code saya:
```driver = webdriver.Chrome('/Applications/chromedriver') ```
```driver.get('https://www.google.com/')```

error yang didapatkan:
 

> /var/folders/kx/5z813z4s6p18qq48txkrn14m0000gn/T/ipykernel_3183/104759495.py:1: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
>   driver = webdriver.Chrome('/Applications/chromedriver')
> ---------------------------------------------------------------------------
> WebDriverException                        Traceback (most recent call last)
> Input In [17], in <cell line: 1>()
> ----> 1 driver = webdriver.Chrome('/Applications/chromedriver')
>       2 driver.get('[https://www.google.com/](https://www.google.com/%3C/span%3E%3Cspan)[ style="color:rgb(175,0,0)">')
> 
> File ~/miniconda3/envs/rhea_da_capstone_web_scraping/lib/python3.8/site-packages/selenium/webdriver/chrome/webdriver.py:70, in WebDriver.__init__(self, executable_path, port, options, service_args, desired_capabilities, service_log_path, chrome_options, service, keep_alive)
>      67 if not service:
>      68     service = Service(executable_path, port, service_args, service_log_path)
> ---> 70 super(WebDriver, self).__init__(DesiredCapabilities.CHROME['browserName'], "goog",
>      71                                 port, options,
>      72                                 service_args, desired_capabilities,
>      73                                 service_log_path, service, keep_alive)
> 
> File ~/miniconda3/envs/rhea_da_capstone_web_scraping/lib/python3.8/site-packages/selenium/webdriver/chromium/webdriver.py:93, in ChromiumDriver.__init__(self, browser_name, vendor_prefix, port, options, service_args, desired_capabilities, service_log_path, service, keep_alive)
>      90 self.service.start()
>      92 try:
> ---> 93     RemoteWebDriver.__init__(
>      94         self,
>      95         command_executor=ChromiumRemoteConnection(
>      96             remote_server_addr=self.service.service_url,
>      97             browser_name=browser_name, vendor_prefix=vendor_prefix,
>      98             keep_alive=keep_alive, ignore_proxy=_ignore_proxy),
>      99         options=options)
>     100 except Exception:
>     101     self.quit()
> 
> File ~/miniconda3/envs/rhea_da_capstone_web_scraping/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py:269, in WebDriver.__init__(self, command_executor, desired_capabilities, browser_profile, proxy, keep_alive, file_detector, options)
>     267 self.file_detector = file_detector or LocalFileDetector()
>     268 self.start_client()
> --> 269 self.start_session(capabilities, browser_profile)
> 
> File ~/miniconda3/envs/rhea_da_capstone_web_scraping/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py:360, in WebDriver.start_session(self, capabilities, browser_profile)
>     357 w3c_caps = _make_w3c_caps(capabilities)
>     358 parameters = {"capabilities": w3c_caps,
>     359               "desiredCapabilities": capabilities}
> --> 360 response = self.execute(Command.NEW_SESSION, parameters)
>     361 if 'sessionId' not in response:
>     362     response = response['value']
> 
> File ~/miniconda3/envs/rhea_da_capstone_web_scraping/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py:425, in WebDriver.execute(self, driver_command, params)
>     423 response = self.command_executor.execute(driver_command, params)
>     424 if response:
> --> 425     self.error_handler.check_response(response)
>     426     response['value'] = self._unwrap_value(
>     427         response.get('value', None))
>     428     return response
> 
> File ~/miniconda3/envs/rhea_da_capstone_web_scraping/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py:247, in ErrorHandler.check_response(self, response)
>     245         alert_text = value['alert'].get('text')
>     246     raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here
> --> 247 raise exception_class(message, screen, stacktrace)
> 
> WebDriverException: Message: unknown error: cannot find Chrome binary
> Stacktrace:
> 0   chromedriver                        0x000000010b5c52c9 chromedriver + 5120713
> 1   chromedriver                        0x000000010b553e33 chromedriver + 4656691
> 2   chromedriver                        0x000000010b143158 chromedriver + 393560
> 3   chromedriver                        0x000000010b1639c8 chromedriver + 526792
> 4   chromedriver                        0x000000010b161446 chromedriver + 517190
> 5   chromedriver                        0x000000010b19b140 chromedriver + 753984
> 6   chromedriver                        0x000000010b195373 chromedriver + 729971
> 7   chromedriver                        0x000000010b16b609 chromedriver + 558601
> 8   chromedriver                        0x000000010b16c635 chromedriver + 562741
> 9   chromedriver                        0x000000010b597b1d chromedriver + 4934429
> 10  chromedriver                        0x000000010b59c295 chromedriver + 4952725
> 11  chromedriver                        0x000000010b5a13cf chromedriver + 4973519
> 12  chromedriver                        0x000000010b59ccba chromedriver + 4955322
> 13  chromedriver                        0x000000010b57737c chromedriver + 4801404
> 14  chromedriver                        0x000000010b5b6c68 chromedriver + 5061736
> 15  chromedriver                        0x000000010b5b6def chromedriver + 5062127
> 16  chromedriver                        0x000000010b5cc5e5 chromedriver + 5150181
> 17  libsystem_pthread.dylib             0x00007ff8139574e1 _pthread_start + 125
> 18  libsystem_pthread.dylib             0x00007ff813952f6b thread_start + 15](https://www.google.com/%3C/span%3E%3Cspan)

sekiranya apa yang salah dari code saya? 

terimakasih

**Comment by onlyphantom** [comment, marked as answer]
The following code

```
driver = webdriver.Chrome(executable_path='/Applications/chromedriver')
```

Returns a warning, saying that the `executable_path` has been deprecated. As of the current version it's not an error yet. Just a warning (meaning, you can ignore it. Code still runs)

> DeprecationWarning: executable_path has been deprecated, please pass in a Service object
> driver = webdriver.Chrome('/Applications/chromedriver')

The following code:

```
driver.get('https://www.google.com/')
``` 
I suspect there is something wrong about your closing quotation (`'`) there. This is from the error message:
> driver.get('[https://www.google.com/](https://www.google.com/%3C/span%3E%3Cspan)[ style="color:rgb(175,0,0)">')
> ...
> (https://www.google.com/%3C/span%3E%3Cspan)

I suspect when you copy it from the browser address bar, it copied along those styles and your editor didn't sanitize / strip those meta away. Try typing `google.com` without copy-pasting and it should work. 

If you suspect there is still an issue, you can switch to another driver and see if it resolves the issue.

```
# if you have Firefox installed
webdriver.Firefox()
# if you're on a Mac
webdriver.Safari()
```
If it does, then go ahead and reinstall the Chrome driver and issue should go away. I have a video to show you how to do that: https://youtu.be/fGBp4Sv4THA
